modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -630,6 +677,114 @@ impl WalletClientModule {
+
+            let outcome_v2 = match outcome_v1 {
+                DepositStateV1::Claimed(tx_info) => DepositStateV2::Claimed {
+                    btc_deposited: bitcoin::Amount::from_sat(
+                        tx_info.btc_transaction.output[tx_info.out_idx as usize].value,
+                    ),
+                    btc_out_point: bitcoin::OutPoint {
+                        txid: tx_info.btc_transaction.txid(),
+                        vout: tx_info.out_idx,
+                    },
+                },
+                DepositStateV1::Failed(error) => DepositStateV2::Failed(error),
+                _ => bail!("Non-final outcome in operation log"),
+            };
+
+            return Ok(UpdateStreamOrOutcome::Outcome(outcome_v2));
+        };
+
+        Ok(operation.outcome_or_updates(&self.client_ctx.global_db(), operation_id, || {
+            let stream_rpc = self.rpc.clone();
+            let stream_cient_ctx = self.client_ctx.clone();
+            let stream_script_pub_key = address.assume_checked().script_pubkey();
+
+            stream! {
+                yield DepositStateV2::WaitingForTransaction;
+
+                retry(
+                    "subscribe script history",
+                    background_backoff(),
+                    || stream_rpc.watch_script_history(&stream_script_pub_key)
+                ).await.expect("Will never give up");
+                let (btc_out_point, btc_deposited) = retry(
+                    "fetch history",
+                    background_backoff(),
+                    || async {
+                        let history = stream_rpc.get_script_history(&stream_script_pub_key).await?;
+                        history.first().and_then(|tx| {
+                            let (out_idx, amount) = tx.output
+                                .iter()
+                                .enumerate()
+                                .find_map(|(idx, output)| (output.script_pubkey == stream_script_pub_key).then_some((idx, output.value)))?;
+                            let txid = tx.txid();
+
+                            Some((
+                                bitcoin::OutPoint {
+                                    txid,
+                                    vout: out_idx as u32,
+                                },
+                                bitcoin::Amount::from_sat(amount)
+                            ))
+                        }).context("No deposit transaction found")
+                    }
+                ).await.expect("Will never give up");
+
+                yield DepositStateV2::WaitingForConfirmation {
+                    btc_deposited,
+                    btc_out_point
+                };
+
+                stream_cient_ctx.module_db().wait_key_exists(&ClaimedPegInKey {
+                    peg_in_index: tweak_idx,
+                    btc_out_point,
+                }).await;
+
+                yield DepositStateV2::Claimed {
```

Comment:
Wondering if we're missing the "await_primary_module_output" (or equivalent) here to actually indicate in the update stream that the e-cash has been issued and added to the balance..? Similar to https://github.com/fedimint/fedimint/blob/5e4247572386518809f5edb86397451a523edf3e/modules/fedimint-ln-client/src/lib.rs#L1624C43-L1624C71

Reply:
Yeah, missed that, but should be easy to fix.